### DISCUSSION TITLE
Instruction to serve the frontend and ui?

### DISCUSSION THREAD

**Discussion by WhistlerSupport** [fabioformosa/quartz-manager#124] [General]
Forgive me if I was just being ignorant. I tried 

- importing quartz-manager-starter-api and quartz-manager-starter-ui into an existing project, or
- running the vanila [existing-quartz](https://github.com/fabioformosa/quartz-manager-use-cases/tree/main/existing-quartz) project (which had quartz-manager-starter-ui imported)

then visited http://localhost:8080/quartz-manager-ui/index.html

however I was unable to see any UI, just a spring Whitelabel Error Page.

it took me to `mvn package -Pbuild-webjar` in the quartz-manager-starter-ui project, (or the whole jar is availabe as download from maven central, just need to dig it out from the .m2 repo)
then extracting the `META-INF\resources\quartz-manager-ui\` resources from the built jar,
then `mvn package` in the project existing-quartz, and put in the quartz-manager-ui resources
and a bunch of other pom manipulation mainly involving copy-dependencies
to finally allow me to run `java -cp existing-quartz-1.0-SNAPSHOT.jar it.fabioformosa.labs.existingquartz.ExistingQuartzApplication` and have it serve the UI in http://localhost:8080/quartz-manager-ui/index.html, as suggested in the [Reach out the UI Console at URL](https://github.com/fabioformosa/quartz-manager/tree/master?tab=readme-ov-file#reach-out-the-ui-console-at-url) section.

Or would it be my tooling issue? I am using Eclipse, not sure if it did not include many of the STS features that would've allowed the UI to be serve easily without much manipulation.

Did I miss some obvious instructions? or can we improve the instruction of serving the frontend UI? I don't mind to help, but I want to know if the correct way lies along what I did.

**Comment by fabioformosa** [comment]
Hi @WhistlerSupport,
you shouldn't build the quartz-manager sourcecode and extracting the built artifact to import it in your project.
As you've noticed, Quartz-Manager is available in the maven central repo. It's not required downloading it and looking for it in the ".m2" folder. 
If your project is based on Maven or Gradle, they will download the required dependencies and they will put them into the classpath.
You just have to specify in your pom.xml (in case you use Maven) or your build.gradle (in case you use Gradle) the dependencies as described here

- https://github.com/fabioformosa/quartz-manager?tab=readme-ov-file#step-1-dependency
- https://github.com/fabioformosa/quartz-manager?tab=readme-ov-file#quartz-manager-starter-ui-lib
